### PR TITLE
LTP: Fix for setsid01 testcase

### PR DESCRIPTION
**Pull request by rsrinivas01**
Test to check the error and trivial conditions in setsid system call. Since fork system call is not supported, the main process creates a thread which in turn creates a child thread. The child thread will try to invoke setsid() and check for the required test condition.

### CODE REVIEW

File: tests/ltp/patches/setsid01.patch
Comment (hukoyu) [inline tests/ltp/patches/setsid01.patch:4]: Please add description for the change. Look at other PRs for sample

File: tests/ltp/patches/setsid01.patch
Comment (hukoyu) [inline tests/ltp/patches/setsid01.patch]: Indentation

**Review by hukoyu — CHANGES_REQUESTED** [review]
@rsrinivas01  please do these:
- Add description to patch file
- Fix the indentation issues
- Use // comments with multiple lines rather than /* */

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: Indentation is changes...

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: same .. indentation issue

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: same .. indentation issue.

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: Please use "SAFE_PTHREAD_CREATE" API instead of pthread_create.  This safe macro is defined in tst_safe_pthread.h.

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: use SAFE_PTHREAD_JOIN instead of pthread_join.

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: Line 49 to 57 looks to be indentation issue.

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: please remove the print statements. use tst_resm with TINFO for information messages.

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: Please use tst_resm API for printing messages.

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch]: use SAFE_PTHREAD_CREATE framework API .

File: tests/ltp/patches/setsid01.patch
Comment (shaikshavali1) [inline tests/ltp/patches/setsid01.patch:206]: do we need another child for infinite loop?

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch:4]: Done

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Done

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Done

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Taken care

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Modified

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Modified accordingly

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Done

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch]: Done

File: tests/ltp/patches/setsid01.patch
Comment (rsrinivas01) [reply tests/ltp/patches/setsid01.patch:206]: Another child is not required as infinite loop is not required for this test.

File: tests/ltp/patches/setsid01.patch
Comment (hukoyu) [reply tests/ltp/patches/setsid01.patch:4]: Where is the patch description? I don't see it.

**Review by hukoyu — CHANGES_REQUESTED** [review]
@rsrinivas01 I don't see the patch description. Also I see lots of code changes a line deleted and added again, I don't understand. Can you get rid of these unnecessary changes. @shaikshavali1 can you review again and  approve if looks good? Without approval, review is not completed.

**Review by rsrinivas01 — COMMENTED** [review]
@shaikshavali1 , @hukoyu  Addressed the alignment issues and other review comments. Please review

**Review by shaikshavali1 — APPROVED** [review]
@rsrinivas01 , @hukoyu    LGTM

**Review by hukoyu — APPROVED** [review]
LGTM

**Pull request merged**
